cmake/BLTPrivateMacros.cmake

```diff
@@ -225,27 +280,95 @@ macro(blt_setup_openmp_target)
-        message( FATAL_ERROR "Building an OpenMP library, but OpenMP is disabled!")
+    if ( NOT ${ENABLE_OPENMP} )
+        message( FATAL_ERROR "Building target '${arg_BUILD_TARGET}' requires OpenMP, but OpenMP is disabled!")
     endif()
 
-    if ( ${arg_USE_OPENMP} )
-        blt_add_target_definitions( TO ${arg_BUILD_TARGET}
+    blt_add_target_definitions( TO ${arg_BUILD_TARGET}
                                 TARGET_DEFINITIONS USE_OPENMP )
-        set_target_properties( ${arg_BUILD_TARGET}
-                               PROPERTIES COMPILE_FLAGS ${OpenMP_CXX_FLAGS} )
-        set_target_properties( ${arg_BUILD_TARGET}
-                               PROPERTIES LINK_FLAGS ${OpenMP_CXX_FLAGS} )
-    endif()
+
+    blt_add_target_compile_flags( TO ${arg_BUILD_TARGET}
+                                  FLAGS ${OpenMP_CXX_FLAGS} )
+
+    blt_add_target_link_flags( TO ${arg_BUILD_TARGET}
+                               FLAGS ${OpenMP_CXX_FLAGS} )
+
 
 endmacro(blt_setup_openmp_target)
 
 ##------------------------------------------------------------------------------
-## update_project_sources( TARGET_SOURCES <souces> )
+## blt_setup_cuda_target( BUILD_TARGET <target> )
+##------------------------------------------------------------------------------
+macro(blt_setup_cuda_target)
+
+    set(options)
+    set(singleValueArgs BUILD_TARGET)
+    set(multiValueArgs)
+
+    ## parse the arguments
+    cmake_parse_arguments(arg "${options}" "${singleValueArgs}" 
+                            "${multiValueArgs}" ${ARGN} )
+
+    if ( NOT DEFINED arg_BUILD_TARGET )
+        message ( FATAL_ERROR "Must provide a BUILD_TARGET argument to the macro")
+    endif()
+
+    if ( NOT ${ENABLE_CUDA} )
+        message( FATAL_ERROR "Building target '${arg_BUILD_TARGET}' requires CUDA, but CUDA is disabled!")
+    endif()
+ 
+    blt_add_target_definitions( TO ${arg_BUILD_TARGET}
+                                TARGET_DEFINITIONS USE_CUDA )
+
+    target_include_directories( ${arg_BUILD_TARGET}
+                                PUBLIC ${CUDA_INCLUDE_DIRS} )
+
+    target_link_libraries( ${arg_BUILD_TARGET}
+                           ${CUDA_LIBRARIES} )
+
+endmacro(blt_setup_cuda_target)
+
+##------------------------------------------------------------------------------
+## blt_setup_cuda_source_properties( TARGET_SOURCES <sources>)
+##------------------------------------------------------------------------------
+macro(blt_setup_cuda_source_properties)
+
+    set(options)
+    set(singleValueArgs )
+    set(multiValueArgs TARGET_SOURCES)
+
+    # Parse the arguments
+    cmake_parse_arguments(arg "${options}" "${singleValueArgs}" 
+                            "${multiValueArgs}" ${ARGN} )
+
+    # Check arguments
+    
+    if ( NOT DEFINED arg_TARGET_SOURCES )
+        message( FATAL_ERROR "Must provide target sources" )
+    endif()
+
+    foreach (_file arg_TARGET_SOURCES)
+        if (${_file} MATCHES "\\.f")
```

Comment:
We need to extend this list to support all the likely Fortran source extensions we might encounter, otherwise it will try and compile the Fortran files with nvcc